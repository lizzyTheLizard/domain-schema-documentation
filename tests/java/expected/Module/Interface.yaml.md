# Interface
![Static Badge](https://img.shields.io/badge/Validator%20Errors-6-red)

This is an interface
```mermaid
classDiagram
namespace Module {
  class _Module_Interface_yaml["Interface"]
  class TestProp["TestProp"]
  class Interface3["Interface3"]
  class OtherObject["OtherObject"]
  class _Module_Object_yaml["Object"]
}
_Module_Interface_yaml <|-- _Module_Object_yaml 
_Module_Interface_yaml <|-- OtherObject 
_Module_Interface_yaml <|-- Interface3 
_Module_Interface_yaml ..> TestProp :testProp
click _Module_Interface_yaml href "./Interface.yaml.html" "Interface"
click TestProp href "./Interface.yaml.html" "Interface"
click Interface3 href "./Interface.yaml.html" "Interface"
click OtherObject href "./Interface.yaml.html" "Interface"
click _Module_Object_yaml href "./Object.yaml.html" "Object"
```


## One Of
1. [Object](./Object.yaml.md)
1. [OtherObject](#OtherObject)
1. [Interface3](#Interface3)


## Properties
| Name | Type | Description |
|------|------|-------------|
| testProp | [TestProp](#TestProp) |  |

## Subschemas
### TestProp (Enum)
This is a test property


| Name | Description |
|------|-------------|
| Test | Test description |
| Test2 | Test2 description |

### Interface3 (Enum)



| Name | Description |
|------|-------------|
| Test | Test description |
| Test2 | Test2 description |

### OtherObject (Object)




| Name | Type | Description |
|------|------|-------------|
| id | String |  |
| wrongType | String |  |
| constV | String<br>"Test" |  |
| constB | Boolean<br>true |  |

## Verification Errors
| Type | Description |
|------|-------------|
| WRONG | Interface &#x27;com.example.module.model.Interface&#x27; is invalid: This is suppose to be an interface but is a &#x27;class&#x27;&#x27; |
| NOT_IN_DOMAIN_MODEL | Value &#x27;Test3&#x27; should not exist in enum &#x27;com.example.module.model.InterfaceInterface3&#x27; |
| MISSING_IN_IMPLEMENTATION | Property &#x27;constV&#x27; is missing in class &#x27;com.example.module.model.InterfaceOtherObject&#x27; |
| MISSING_IN_IMPLEMENTATION | Property &#x27;constB&#x27; is missing in class &#x27;com.example.module.model.InterfaceOtherObject&#x27; |
| WRONG | Property &#x27;wrongType&#x27; has type &#x27;Integer&#x27; in class &#x27;com.example.module.model.InterfaceOtherObject&#x27; but should have type &#x27;String&#x27; |
| NOT_IN_DOMAIN_MODEL | Property &#x27;wrongProperty&#x27; should not exist in class &#x27;com.example.module.model.InterfaceOtherObject&#x27; |

## Links
1. [Java-File](./java/Interface.java)
1. [Java-File (TestProp)](./java/InterfaceTestProp.java)
1. [Java-File (Interface3)](./java/InterfaceInterface3.java)
1. [Java-File (OtherObject)](./java/InterfaceOtherObject.java)
