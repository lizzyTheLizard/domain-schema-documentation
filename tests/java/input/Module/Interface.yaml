$id: /Module/Interface.yaml
title: Interface
description: This is an interface
type: object
x-schema-type: Other
oneOf:
  - $ref: ./Object.yaml
  - $ref: '#/definitions/OtherObject'
  - type: string
    enum:
      - "Test"
      - "Test2"
    x-enum-description:
      Test: "Test description"
      Test2: "Test2 description"
properties:
  testProp: 
    type: string
    description: This is a test property
    x-enum-description:
      Test: "Test description"
      Test2: "Test2 description"
    enum:
      - "Test"
      - "Test2"
definitions:
  OtherObject:
    type: object
    properties:
      id:
        type: string
      wrongType:
        type: string
      constV:
        type: string
        const: "Test"
      constB:
        type: boolean
        const: true