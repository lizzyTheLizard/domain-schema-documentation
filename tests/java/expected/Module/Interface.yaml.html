<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="Mark Otto, Jacob Thornton, and Bootstrap contributors">
    <meta name="generator" content="Hugo 0.84.0">
    <title>Test Application</title>
    <link rel="canonical" href="https://getbootstrap.com/docs/5.0/examples/navbar-static/">
    <link rel=stylesheet href=https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/css/pretty-print-json.css>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet">
    <meta name="theme-color" content="#7952b3">
    <style>
    .alert ul {
      margin-bottom: 0;
    }
    .mermaid{
      height: 300px;
      border: 1px solid #000;
    }
    .mermaid pre {
      display: none;
    }
    section {
      margin-bottom: 50px;
    }
    .title-badge {
      font-size: 0.3em;
      vertical-align: top;
    }
    .json {
      border: 1px solid black;
      background-color: lightgray;
      padding: 10px;
     }
    </style>
  </head>
  <body>
    <main class="container">
      <nav aria-label="breadcrumb">
        <ol class="breadcrumb">
          <li class="breadcrumb-item"><a href="../../index.html">Application</a></li>
          <li class="breadcrumb-item"><a href="./index.html">Module</a></li>
          <li class="breadcrumb-item active" aria-current="page">Interface</li>
        </ol>
      </nav>
      
      <h1>Interface</h1>
      <img alt="Static Badge" src="https://img.shields.io/badge/Validator%20Errors-6-red">
      
      
      <section>
        <p>This is an interface</p>
        <div class="mermaid">
          <pre class="mermaidDefinition">
            classDiagram
      namespace Module {
        class _Module_Interface_yaml["Interface"]
        class TestProp["TestProp"]
        class Interface3["Interface3"]
        class OtherObject["OtherObject"]
        class _Module_Object_yaml["Object"]
      }
      _Module_Interface_yaml <|-- _Module_Object_yaml 
      _Module_Interface_yaml <|-- OtherObject 
      _Module_Interface_yaml <|-- Interface3 
      _Module_Interface_yaml ..> TestProp :testProp
      click _Module_Interface_yaml href "./Interface.yaml.html" "Interface"
      click TestProp href "./Interface.yaml.html" "Interface"
      click Interface3 href "./Interface.yaml.html" "Interface"
      click OtherObject href "./Interface.yaml.html" "Interface"
      click _Module_Object_yaml href "./Object.yaml.html" "Object"
          </pre>
        </div>
      </section>
      
      <section>
        <h2>One Of</h2>
        <ul>
          <li><a href="./Object.yaml.html">Object</a></li>
          <li><a href="#OtherObject">OtherObject</a></li>
          <li><a href="#Interface3">Interface3</a></li>
        </ul>
      </section>
      
      
      <section>
        <h2>Properties</h2>
        <table class="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>testProp</td>
              <td><a href="#TestProp">TestProp</a></td>
              <td></td>
            </tr>
          </tbody>
        </table>
      </section>
      
      
      <section>
        <h2>Verification Errors</h2>
        <table class="table">
          <thead>
            <tr>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>WRONG</td>
              <td>Interface &#x27;com.example.module.model.Interface&#x27; is invalid: This is suppose to be an interface but is a &#x27;class&#x27;&#x27;</td>
            </tr>
            <tr>
              <td>NOT_IN_DOMAIN_MODEL</td>
              <td>Value &#x27;Test3&#x27; should not exist in enum &#x27;com.example.module.model.InterfaceInterface3&#x27;</td>
            </tr>
            <tr>
              <td>MISSING_IN_IMPLEMENTATION</td>
              <td>Property &#x27;constV&#x27; is missing in class &#x27;com.example.module.model.InterfaceOtherObject&#x27;</td>
            </tr>
            <tr>
              <td>MISSING_IN_IMPLEMENTATION</td>
              <td>Property &#x27;constB&#x27; is missing in class &#x27;com.example.module.model.InterfaceOtherObject&#x27;</td>
            </tr>
            <tr>
              <td>WRONG</td>
              <td>Property &#x27;wrongType&#x27; has type &#x27;Integer&#x27; in class &#x27;com.example.module.model.InterfaceOtherObject&#x27; but should have type &#x27;String&#x27;</td>
            </tr>
            <tr>
              <td>NOT_IN_DOMAIN_MODEL</td>
              <td>Property &#x27;wrongProperty&#x27; should not exist in class &#x27;com.example.module.model.InterfaceOtherObject&#x27;</td>
            </tr>
          </tbody>
        </table>
      </section>
      
      <section>
        <h2>Links</h2>
        <ul>
          <li><a href="./java/Interface.java">Java-File</a></li>
          <li><a href="./java/InterfaceTestProp.java">Java-File (TestProp)</a></li>
          <li><a href="./java/InterfaceInterface3.java">Java-File (Interface3)</a></li>
          <li><a href="./java/InterfaceOtherObject.java">Java-File (OtherObject)</a></li>
        </ul>
      </section>
      
      
      <h2>Subschemas</h2>
      <a id="TestProp"></a><h3>TestProp   (Enum)  </h3>
      <section>
        <p>This is a test property</p>
        <table class="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Test</td>
              <td>Test description</td>
            </tr>
            <tr>
              <td>Test2</td>
              <td>Test2 description</td>
            </tr>
          </tbody>
        </table>
      
      </section>
      <a id="Interface3"></a><h3>Interface3   (Enum)  </h3>
      <section>
        <p></p>
        <table class="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>Test</td>
              <td>Test description</td>
            </tr>
            <tr>
              <td>Test2</td>
              <td>Test2 description</td>
            </tr>
          </tbody>
        </table>
      
      </section>
      <a id="OtherObject"></a><h3>OtherObject    (Object)</h3>
      <section>
        <p></p>
      
        <table class="table">
          <thead>
            <tr>
              <th>Name</th>
              <th>Type</th>
              <th>Description</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>id</td>
              <td>String</td>
              <td></td>
            </tr>
            <tr>
              <td>wrongType</td>
              <td>String</td>
              <td></td>
            </tr>
            <tr>
              <td>constV</td>
              <td>String<br>"Test"</td>
              <td></td>
            </tr>
            <tr>
              <td>constB</td>
              <td>Boolean<br>true</td>
              <td></td>
            </tr>
          </tbody>
        </table>
      </section>
      
    </main>
    <script src="https://cdn.jsdelivr.net/npm/svg-pan-zoom@3.5.0/dist/svg-pan-zoom.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/pretty-print-json@3.0/dist/pretty-print-json.min.js"></script>
    <script type="module">
      import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';

      function prettifyJson(){
        const elements = document.querySelectorAll('.json');
        for(const element of elements) {
          element.innerHTML = prettyPrintJson.toHtml(JSON.parse(element.innerText));
        }
      }

      function drawDiagrams() {
        const elements = document.querySelectorAll('.mermaid');
        elements.forEach((element, i) =>drawDiagram(element, i));
      }

      async function drawDiagram(element, i) {
        const graphDefinition = element.querySelector(".mermaidDefinition")
        const {svg} = await mermaid.render('mySvg_' + i, graphDefinition.innerText);
        const preview = createMermaidPreview(element, svg);
        const modal = createMermaidModal();
        //Do this with a slight delay to take link in mermaid diagram take precedence
        element.addEventListener('click', () => setTimeout(() => showMermaidModal(preview,modal), 30));
      }

      function createMermaidPreview(element, svg) {
        const preview = document.createElement('div');
        preview.style.height = '100%';
        preview.style.width = '100%';
        preview.innerHTML = svg.replace(/[ ]*max-width:[ 0-9\.]*px;/i , '');
        preview.querySelector("svg").style.height = '100%';
        preview.querySelector("svg").style.width = '100%';
        element.appendChild(preview);
        return preview;
      }

      function createMermaidModal() {
        const modal = document.createElement('div');
        modal.innerHTML =
            '<div class="modal" tabindex="-1">\n' +
            '  <div class="modal-dialog modal-fullscreen">\n' +
            '    <div class="modal-content">\n' +
            '      <div class="modal-header">\n' +
            '        <button type="button" class="btn-close" data-bs-dismiss="modal" aria-label="Close"></button>\n' +
            '      </div>\n' +
            '      <div class="modal-body"></div>\n' +
            '    </div>\n' +
            '  </div>\n' +
            '</div>';
        document.body.appendChild(modal);
        return modal;
      }

      function showMermaidModal(preview, modal) {
        const modalBody = modal.querySelector('.modal-body');
        modalBody.innerHTML = preview.innerHTML
        const modalDiv = modal.querySelector('.modal');
        const svg = modalBody.querySelector('svg');
        new bootstrap.Modal(modalDiv).show();
        svgPanZoom(svg, {
          zoomEnabled: true,
          controlIconsEnabled: true,
          fit: true,
          center: true
        })
      }

      mermaid.initialize({ startOnLoad: false,securityLevel: 'loose'});
      drawDiagrams();
      prettifyJson();
    </script>
  </body>
</html>
